song.form.html

```diff
@@ -0,0 +1,18 @@
+<h2>Add a Song</h2>
+
+<input type="text" data-ng-model="song.title" />
+<label for="title">Title</label>
+
+<input type="text" data-ng-model="song.artist" />
+<label for="artist">Artists</label>
+
+<input type="text" data-ng-model="song.genre" />
+<label for="genre">Genre</label>
+
+<input type="text" data-ng-model="song.audio_url" />
+<label for="audio_url">URL to audio file</label>
+
+<input type="text" data-ng-model="song.album_art" />
+<label for="album_art">URL to album cover</label>
+
+<button type="button" data-ng-click="create()">Submit</button>
```

Comment:
so there are a couple ways you can include the edit button here as well as create. i've done it where i include an `ng-if = song.id` on the edit button, which says if the song exists, than show that, and the opposite for create. or, you could add in something like `data-form-type`, and input that into the directive tag